test/unit/workers/worker_base_test.rb

```diff
@@ -0,0 +1,27 @@
+class WorkerBaseTest < ActiveSupport::TestCase
+  def self.worker_has_run!
+  end
+
+  class MyWorker < WorkerBase
+    def perform
+      WorkerBaseTest.worker_has_run!
+    end
+  end
+
+  test ".perform_async runs the job" do
+    self.class.expects(:worker_has_run!)
+    MyWorker.perform_async
+  end
+
+  test ".perform_async_in_queue runs the job in the specified queue" do
```

Comment:
I tried this but the redis API does not work with sidekiq test modes. The call to `Sidekiq::Queue` attempts to connect to redis regardless of whether sidekiq `inline` or `fake` mode is active.

Dan and I discussed this approach to testing (by setting expectations on the call to `client_push`), we felt that since this is effectively testing integration with sidekiq, we're ok to have a test which is tightly coupled to Sidekiq's interface. If Sidekiq's interface changes, we would want it to break.